<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MultiMaterialSimulation</class>
 <widget class="qMRMLWidget" name="MultiMaterialSimulation">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>486</width>
    <height>556</height>
   </rect>
  </property>
  <property name="SlicerParameterName" stdset="0">
   <string>simulationModelNode</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QFormLayout" name="formLayout_2">
     <item row="0" column="0">
      <widget class="QLabel" name="simulationModelNodeLabel">
       <property name="text">
        <string>Simulation model:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="qMRMLNodeComboBox" name="SimulationModelComboBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="nodeTypes">
          <stringlist notr="true">
           <string>vtkMRMLModelNode</string>
          </stringlist>
         </property>
         <property name="showHidden">
          <bool>false</bool>
         </property>
         <property name="hideChildNodeTypes">
          <stringlist notr="true"/>
         </property>
         <property name="noneEnabled">
          <bool>true</bool>
         </property>
         <property name="addEnabled">
          <bool>false</bool>
         </property>
         <property name="removeEnabled">
          <bool>false</bool>
         </property>
         <property name="editEnabled">
          <bool>false</bool>
         </property>
         <property name="renameEnabled">
          <bool>false</bool>
         </property>
         <property name="interactionNodeSingletonTag">
          <string notr="true"/>
         </property>
         <property name="SlicerParameterName" stdset="0">
          <string>simulationModelNode</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>31</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="simulationModelNodeLabel_2">
       <property name="text">
        <string>Shell static model:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_11">
       <item>
        <widget class="qMRMLNodeComboBox" name="ShellStaticModelComboBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="nodeTypes">
          <stringlist notr="true">
           <string>vtkMRMLModelNode</string>
          </stringlist>
         </property>
         <property name="showHidden">
          <bool>false</bool>
         </property>
         <property name="hideChildNodeTypes">
          <stringlist notr="true"/>
         </property>
         <property name="noneEnabled">
          <bool>true</bool>
         </property>
         <property name="addEnabled">
          <bool>false</bool>
         </property>
         <property name="removeEnabled">
          <bool>false</bool>
         </property>
         <property name="editEnabled">
          <bool>false</bool>
         </property>
         <property name="renameEnabled">
          <bool>false</bool>
         </property>
         <property name="interactionNodeSingletonTag">
          <string notr="true"/>
         </property>
         <property name="SlicerParameterName" stdset="0">
          <string>shellStaticModelNode</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>31</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="simulationModelNodeLabel_3">
       <property name="text">
        <string>Shell moving model:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_13">
       <item>
        <widget class="qMRMLNodeComboBox" name="ShellMovingModelComboBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="nodeTypes">
          <stringlist notr="true">
           <string>vtkMRMLModelNode</string>
          </stringlist>
         </property>
         <property name="showHidden">
          <bool>false</bool>
         </property>
         <property name="hideChildNodeTypes">
          <stringlist notr="true"/>
         </property>
         <property name="noneEnabled">
          <bool>true</bool>
         </property>
         <property name="addEnabled">
          <bool>false</bool>
         </property>
         <property name="removeEnabled">
          <bool>false</bool>
         </property>
         <property name="editEnabled">
          <bool>false</bool>
         </property>
         <property name="renameEnabled">
          <bool>false</bool>
         </property>
         <property name="interactionNodeSingletonTag">
          <string notr="true"/>
         </property>
         <property name="SlicerParameterName" stdset="0">
          <string>shellMovingModelNode</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_6">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>31</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="boundaryROILabel">
       <property name="text">
        <string>Boundary ROI:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="qMRMLNodeComboBox" name="boundaryROIComboBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="nodeTypes">
          <stringlist notr="true">
           <string>vtkMRMLMarkupsROINode</string>
          </stringlist>
         </property>
         <property name="hideChildNodeTypes">
          <stringlist notr="true"/>
         </property>
         <property name="noneEnabled">
          <bool>true</bool>
         </property>
         <property name="addEnabled">
          <bool>false</bool>
         </property>
         <property name="interactionNodeSingletonTag">
          <string notr="true"/>
         </property>
         <property name="selectNodeUponCreation">
          <bool>true</bool>
         </property>
         <property name="SlicerParameterName" stdset="0">
          <string>boundaryROI</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="addBoundaryROIPushButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>31</width>
           <height>23</height>
          </size>
         </property>
         <property name="text">
          <string>+</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="gravityVectorLabel">
       <property name="text">
        <string>Gravity Vector:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="qMRMLNodeComboBox" name="gravityVectorComboBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="nodeTypes">
          <stringlist notr="true">
           <string>vtkMRMLMarkupsLineNode</string>
          </stringlist>
         </property>
         <property name="hideChildNodeTypes">
          <stringlist notr="true"/>
         </property>
         <property name="noneEnabled">
          <bool>true</bool>
         </property>
         <property name="addEnabled">
          <bool>false</bool>
         </property>
         <property name="interactionNodeSingletonTag">
          <string notr="true"/>
         </property>
         <property name="SlicerParameterName" stdset="0">
          <string>gravityVector</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="gravityMagnitudeSpinBox">
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>9999</number>
         </property>
         <property name="stepType">
          <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
         </property>
         <property name="value">
          <number>0</number>
         </property>
         <property name="SlicerParameterName" stdset="0">
          <string>gravityMagnitude</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="addGravityVectorPushButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>31</width>
           <height>23</height>
          </size>
         </property>
         <property name="text">
          <string>+</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="simulationModelNodeLabel_4">
       <property name="text">
        <string>LVAD Transform:</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_14">
       <item>
        <widget class="qMRMLNodeComboBox" name="LVADTransformComboBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="nodeTypes">
          <stringlist notr="true">
           <string>vtkMRMLTransformNode</string>
          </stringlist>
         </property>
         <property name="showHidden">
          <bool>false</bool>
         </property>
         <property name="hideChildNodeTypes">
          <stringlist notr="true"/>
         </property>
         <property name="noneEnabled">
          <bool>true</bool>
         </property>
         <property name="addEnabled">
          <bool>false</bool>
         </property>
         <property name="removeEnabled">
          <bool>false</bool>
         </property>
         <property name="editEnabled">
          <bool>false</bool>
         </property>
         <property name="renameEnabled">
          <bool>false</bool>
         </property>
         <property name="interactionNodeSingletonTag">
          <string notr="true"/>
         </property>
         <property name="SlicerParameterName" stdset="0">
          <string>LVADTransformNode</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="addLVADTransformPushButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>31</width>
           <height>23</height>
          </size>
         </property>
         <property name="text">
          <string>+</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="recordingSequenceLabel">
       <property name="text">
        <string>Rec. Sequence:</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="qMRMLNodeComboBox" name="recordingSequenceComboBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="nodeTypes">
          <stringlist notr="true">
           <string>vtkMRMLSequenceNode</string>
          </stringlist>
         </property>
         <property name="hideChildNodeTypes">
          <stringlist notr="true"/>
         </property>
         <property name="noneEnabled">
          <bool>true</bool>
         </property>
         <property name="addEnabled">
          <bool>false</bool>
         </property>
         <property name="interactionNodeSingletonTag">
          <string notr="true"/>
         </property>
         <property name="SlicerParameterName" stdset="0">
          <string>sequenceNode</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="addRecordingSequencePushButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>31</width>
           <height>23</height>
          </size>
         </property>
         <property name="text">
          <string>+</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_7">
     <item>
      <widget class="QPushButton" name="startSimulationPushButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Start Simulation</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="stopSimulationPushButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Stop Simulation</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="dtLabel">
         <property name="text">
          <string>dt:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QDoubleSpinBox" name="dtSpinBox">
         <property name="decimals">
          <number>3</number>
         </property>
         <property name="singleStep">
          <double>0.001000000000000</double>
         </property>
         <property name="stepType">
          <enum>QAbstractSpinBox::DefaultStepType</enum>
         </property>
         <property name="value">
          <double>0.001000000000000</double>
         </property>
         <property name="SlicerParameterName" stdset="0">
          <string>dt</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QLabel" name="currentStepLabel">
         <property name="text">
          <string>Current Step:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="currentStepSpinBox">
         <property name="maximum">
          <number>999999999</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
         <property name="SlicerParameterName" stdset="0">
          <string>currentStep</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QFormLayout" name="formLayout_3">
       <item row="0" column="0">
        <widget class="QLabel" name="totalStepsLabel">
         <property name="text">
          <string>Total Steps:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="totalStepsSpinBox">
         <property name="minimum">
          <number>-1</number>
         </property>
         <property name="maximum">
          <number>999999999</number>
         </property>
         <property name="value">
          <number>-1</number>
         </property>
         <property name="SlicerParameterName" stdset="0">
          <string>totalSteps</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>MultiMaterialSimulation</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>SimulationModelComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>182</x>
     <y>155</y>
    </hint>
    <hint type="destinationlabel">
     <x>257</x>
     <y>21</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MultiMaterialSimulation</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>recordingSequenceComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>264</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>215</x>
     <y>141</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MultiMaterialSimulation</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>boundaryROIComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>264</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>215</x>
     <y>51</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MultiMaterialSimulation</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>gravityVectorComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>264</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>183</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MultiMaterialSimulation</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>LVADTransformComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>288</x>
     <y>178</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MultiMaterialSimulation</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>ShellMovingModelComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>288</x>
     <y>84</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MultiMaterialSimulation</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>ShellStaticModelComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>288</x>
     <y>53</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
